packages/eas-cli/src/commands/secret/create.ts

```diff
@@ -95,17 +106,46 @@ export default class EnvironmentSecretCreate extends EasCommand {
     }
 
     if (!secretValue) {
-      const validationMessage = 'Secret value may not be empty.';
-
       ({ secretValue } = await promptAsync({
         type: 'text',
         name: 'secretValue',
-        message: 'Secret value:',
-        validate: value => (value ? true : validationMessage),
+        message: 'Secret value (or local file path):',
+        // eslint-disable-next-line async-protect/async-suffix
+        validate: async secretValue => {
+          if (!secretValue) {
+            return 'Secret value may not be empty.';
+          }
+          if (type === SecretType.FILE) {
+            const secretFilePath = path.resolve(secretValue);
+            if (!(await fs.pathExists(secretFilePath))) {
+              return 'Secret file does not exist.';
+            }
+          }
+          return true;
+        },
       }));
+    }
+
+    assert(secretValue);
 
-      if (!secretValue) {
-        throw new Error(validationMessage);
+    let secretFilePath: string | undefined;
+    if (type !== SecretType.STRING) {
+      secretFilePath = path.resolve(secretValue);
+      if (!(await fs.pathExists(secretFilePath))) {
+        if (type === SecretType.FILE) {
+          throw new Error(`File "${secretValue}" does not exist`);
+        }
+        type = SecretType.STRING;
+      } else {
+        const confirmed = await confirmAsync({
+          message: `This will create a file secret with the current contents of ${secretValue}. Do you want to continue?`,
```

Comment:
I don't know if this sounds right.  If the user says "yes" we read the file, otherwise, we use the value as is.

Reply:
I think we should stop the process here if they say no. It doesn't make sense to proceed with an unintended action if the user said they don't want to continue

Reply:
i think @FiberJW's suggestion is reasonable - we can exit and say something like 'if you want to create a secret with that value then pass in the `--type string` flag'

Reply:
I think the point raised is valid. We already went back and forth about 'detecting' a file, and that where this warning came from. But if we are wondering what happens when you select no, so will the user. This makes me rethink if `--type file` shouldn't always be explicitly passed if you want to create a file secret, and drop the 'detect-a-file' logic altogether.

Reply:
I just think we need two clear, predictable paths. 

If no `--type` flag is passed, try to auto-detect a file and prompt if it was meant to be a file or a string. In this case, we shouldn't do a yes/no or continue/quit prompt. We should do a "is this a file or string" prompt and just fail if they select file, but the file doesn't exist.

If `--type=file` is passed, only try to create a file secret, don't fall back on making a string secret (and the inverse for if a string flag was specified)

Reply:
The most predictable path is just to ask if no flag was passed. No auto-detection needed first. Just ask what type of secret. Like we do with other commands if a flag isn't passed in. And fail if file not found (in both cases) obviously.

Reply:
alright let's just prompt and not try to detect

Reply:
Ok 🙃

Reply:
I changed it so we always display a prompt if `--type` is not passed.